derive_client/data_types/models.py

```diff
@@ -66,3 +67,29 @@ class NonMintableTokenData(TokenData):
 class DeriveAddresses(BaseModel):
     model_config = ConfigDict(arbitrary_types_allowed=True)
     chains: dict[ChainID, dict[Currency, MintableTokenData | NonMintableTokenData]]
+
+
+@dataclass
+class TxResult:
+    tx_hash: str
+    tx_receipt: AttributeDict | None
+    exception: Exception | None
+
+    @property
+    def status(self) -> TxStatus:
+        if self.tx_receipt is not None:
+            return TxStatus(int(self.tx_receipt.status))  # ∈ {0, 1} (EIP-658)
+        if isinstance(self.exception, TimeoutError):
+            return TxStatus.PENDING
+        return TxStatus.ERROR
```

Comment:
the `BridgeClient` returns this from it's methods to deposit / withdraw with help of the `send_and_confirm_tx` function